app/server.js

```diff
@@ -22,10 +55,109 @@ app.use((err, req, res, next) => {
 // Swagger
 app.use('/api-docs', swaggerUi.serve, swaggerUi.setup(swaggerDocument));
 
+// Serve static files from 'public' directory
+app.use(express.static(path.join(__dirname, '../public')));
+
+// Route for root path
+app.get('/', (req, res) => {
+  res.send('../index.html');
+});
+
+// Routes for /brands, /products, and /users
+app.get('/brands', (req, res) => {
+  const brandNames = brands.map(brand => brand.name);
+  res.json(brandNames);
+});
+
+app.get('/brands/:name', (req, res) => {
+  // NO SYMBOLS & NO SPACE
+  const brandName = req.params.name.toLowerCase(); //Oakley
+  const brand = brands.find(brand => brand.name.toLowerCase() === brandName); //Oakley
+  const brandId = brand.id; // 1
+  const product = products.filter(product => product.categoryId === brandId);
+  if (brand) {
+    res.json(product);
+  } else {
+    res.status(404).send('Products by brand not found');
+  };
+});
+
+app.get('/products', (req, res) => {
+	const productNames = products.map(product => product.name);
+	res.json(productNames);
+});
+
+app.get('/products/:name', (req, res) => {
+  // NO SYMBOLS & NO SPACE
+  const productName = req.params.name.toLowerCase();
+  const product = products.find(product => product.name.toLowerCase() === productName);
+  const productDetails = {
+    name: product.name,
+    description: product.description,
+    price: product.price,
+    imageUrls: product.imageUrls
+  };
+  if (product) {
+    res.json(productDetails);
+  } else {
+    res.status(404).send('Product not found');
+  };
+});
+
+// AUTHENTICATED ROUTES
+app.get('/login', authenticate, (req, res) => {
+  res.status(200).send(`Let's shop!`);
+});
+
+app.get('/users', authenticate, (req, res) => {
+  const userNames = users.map(user => user.name.first);
+  res.status(200).json(userNames);
+});
+
+app.get('/:name', authenticate, (req, res) => {
+  const userName = req.params.name.toLowerCase();
+  const user = users.find(user => user.name.first.toLowerCase() === userName);
+
+  if (user) {
+    const userCart = user.cart;
+    res.status(200).json(userCart);
+  } else {
+    res.status(401).send('Unauthorized');
```

Comment:
this is not accurate, if you didnt find the user, means that the user was not found (400), not that the request is unauthorized